sdk/inc/azure/core/az_credentials_common.h

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ *
+ * @brief Azure credentials common definitions.
+ *
+ * @note You MUST NOT use any symbols (macros, functions, structures, enums, etc.)
+ * prefixed with an underscore ('_') directly in your application code. These symbols
+ * are part of Azure SDK's internal implementation; we do not document these symbols
+ * and they are subject to change in future versions of the SDK which would break your code.
+ */
+
+#ifndef _az_CREDENTIALS_COMMON_H
+#define _az_CREDENTIALS_COMMON_H
+
+#include <azure/core/az_http_transport.h>
+
+#include <stddef.h>
+
+#include <azure/core/_az_cfg_prefix.h>
+
+/**
+ * @brief Equivalent to no credential (`NULL`).
+ */
+#define AZ_CREDENTIAL_ANONYMOUS NULL
+
+/**
+ * @brief Function callback definition as a contract to be implemented for a credential to set
+ * authentication scopes when it is supported by the type of the credential.
+ */
+typedef AZ_NODISCARD az_result (
+    *_az_credential_set_scopes_fn)(void* ref_credential, az_span scopes);
+
+/**
+ * @brief Pointer to a credential implementation.
+ */
+typedef void* az_platform_credential;
+
+/**
+ * @brief Defines the supported credential types.
+ */
+typedef enum
```

Comment:
This should be removed.  Core does not know about credential types it only knows about the AccessToken and the TokenCredential type.  Types expanding on TokenCredential are identity constructs.

Reply:
You can use the C++ impl to reference the Azure SDK design patterns for credentials.
https://github.com/Azure/azure-sdk-for-cpp/blob/main/sdk/core/azure-core/inc/azure/core/credentials/credentials.hpp

Reply:
The change is based from an offline discussion we had with @LarryOsterman . Sharing the summary here:

> We will create pal_x509certifiate which represents the certificate for the selected platform (e.g. MosquittoMQTT).
> - az_credential  will become a public type.  
> - We will add a constructor (_init function) that accepts a pal_x509certificate 
> - A separate constructor will be used for SAS az_credentials

Reply:
@RickWinter what would be the recommended way to add a public API for X.509 credentials?
Should we create our own `az_iot_` type instead?

Reply:
Hi @RickWinter, I've gone ahead and removed `az_credentials_common.h` and moved my changes to `az_iot_certificate_x509`.